src/components/dialog-full-screen/dialog-full-screen.test.tsx

```diff
@@ -0,0 +1,432 @@
+
+  expect(screen.getByRole("dialog")).toHaveAccessibleDescription("Subtitle");
+});
+
+test("when the `title` prop is a React element and the `aria-labelledby` prop is provided, it is passed to the dialog container", () => {
+  render(
+    <>
+      <h2 id="test-id">custom title</h2>
+      <DialogFullScreen open title={<h2>test</h2>} aria-labelledby="test-id" />
+    </>
+  );
+
+  expect(screen.getByRole("dialog")).toHaveAccessibleName("custom title");
+});
+
+test("when the `subtitle` prop is a React element and the `aria-describedby` prop is provided, it is passed to the dialog container", () => {
+  render(
+    <>
+      <h2 id="test-id">custom description</h2>
+      <DialogFullScreen
+        open
+        title="test"
+        subtitle={<p>custom subtitle</p>}
+        aria-describedby="test-id"
+      />
+    </>
+  );
+
+  expect(screen.getByRole("dialog")).toHaveAccessibleDescription(
+    "custom description"
+  );
+});
+
+test("the role prop is passed to the dialog element", () => {
+  render(
+    <DialogFullScreen open role="alert">
+      dialog content
+    </DialogFullScreen>
+  );
+
+  expect(screen.getByRole("alert")).toHaveTextContent("dialog content");
+});
+
+test("when the `title` prop is not passed and the `aria-label` prop is provided, it is passed to the dialog container as its accessible name", () => {
+  render(<DialogFullScreen open aria-label="custom label" />);
+
+  expect(screen.getByRole("dialog")).toHaveAccessibleName("custom label");
+});
+
+test("the onCancel callback should be called when the close button is clicked", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+  const onCancel = jest.fn();
+  render(<ControlledDialog onCancel={onCancel} />);
+
+  await user.click(screen.getByRole("button", { name: "Close" }));
+  jest.runAllTimers();
+  expect(screen.queryByRole("dialog")).not.toBeInTheDocument();
+  expect(onCancel).toHaveBeenCalledTimes(1);
+});
+
+test("the onCancel callback should be called when the enter key is pressed", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+  const onCancel = jest.fn();
+  render(<ControlledDialog onCancel={onCancel} />);
+
+  screen.getByRole("button", { name: "Close" }).focus();
+  await user.keyboard("{Enter}");
+  jest.runAllTimers();
+  expect(screen.queryByRole("dialog")).not.toBeInTheDocument();
+  expect(onCancel).toHaveBeenCalledTimes(1);
+});
+
+test("the onCancel callback should not be called when a non-Enter is pressed", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+  const onCancel = jest.fn();
+  render(<ControlledDialog onCancel={onCancel} />);
+
+  screen.getByRole("button", { name: "Close" }).focus();
+  await user.keyboard("a");
+  jest.runAllTimers();
+  expect(screen.getByRole("dialog")).toBeVisible();
+  expect(onCancel).not.toHaveBeenCalled();
+});
```

Comment:
**suggestion**: similarly what do you think about using `waitFor` here instead?: 

```suggestion

  expect(screen.getByRole("dialog")).toBeVisible();
  await waitFor(() => {
    expect(onCancel).not.toHaveBeenCalled();
  });
```

Reply:
I'm nervous about the `waitFor` approach here, for the same reason as I explained in [this comment](https://github.com/Sage/carbon/pull/6915#discussion_r1740869722) - waiting for something to be visible that's already visible isn't going to fail even if we were to accidentally cause the dialog to close (after a small delay) in response to a non-Enter press.

I realise your comment is about the `not.toHaveBeenCalled` assertion rathre than the `toBeVisible` one - but in that case it seems strange to me to use `waitFor`, as you can't wait for something *not* to be have been called, as once the function has been called that status is never going to change, no matter how long you wait!

So I think I'll leave this test as it is, with an explanatory comment as I explained in the reply I just linked to